dev/tests/api-functional/testsuite/Magento/GraphQl/Quote/Customer/SetShippingMethodsOnCartTest.php

```diff
@@ -116,6 +117,33 @@ public function testSetMultipleShippingMethods()
         $this->markTestIncomplete('https://github.com/magento/graphql-ce/issues/423');
     }
 
+    /**
+     * @magentoApiDataFixture Magento/Checkout/_files/quote_with_shipping_method.php
+     */
+    public function testReSetShippingMethod()
+    {
+        $maskedQuoteId = $this->getMaskedQuoteIdByReversedQuoteId('test_order_1');
+        $methodCode = 'flatrate';
+        $carrierCode = 'flatrate';
+        $quote = $this->getQuoteByReversedQuoteId('test_order_1');
+        $shippingAddressId = $quote->getShippingAddress()->getId();
+        
+        $query = $this->prepareMutationQuery($maskedQuoteId, $methodCode, $carrierCode, $shippingAddressId);
+        $response = $this->graphQlQuery($query, [], '', $this->getHeaderMap());
+
+        self::assertArrayHasKey('setShippingMethodsOnCart', $response);
+        self::assertArrayHasKey('cart', $response['setShippingMethodsOnCart']);
+        self::assertArrayHasKey('shipping_addresses', $response['setShippingMethodsOnCart']['cart']);
+        foreach ($response['setShippingMethodsOnCart']['cart']['shipping_addresses'] as $address) {
```

Comment:
If we use `foreach` that means we have a few shipping addresses for Cart.
Need to check why don't we have an exception `Multishipping is not supported` with this preconditions
Thanks

Reply:
Do you mean to change `shipping_addresses` field of Cart type to `shipping_address`
 and to throw  `Multishipping is not supported ` exception in \Magento\QuoteGraphQl\Model\Resolver\ShippingAddresses resolver if there is more than one shipping address?

Reply:
Do you mean to change `shipping_addresses` field of Cart type to `shipping_address`
 and to throw  `Multishipping is not supported ` exception in \Magento\QuoteGraphQl\Model\Resolver\ShippingAddresses resolver if there is more than one shipping address?